galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/notif/VWarningNotification.kt

```diff
@@ -18,67 +18,37 @@
 package org.kopi.galite.ui.vaadin.notif
 
 import org.kopi.galite.ui.vaadin.base.LocalizedProperties
-import org.kopi.galite.ui.vaadin.base.VInputButton
 
-import com.vaadin.flow.component.ShortcutEvent
-import com.vaadin.flow.component.html.H3
-import com.vaadin.flow.component.html.Span
-import com.vaadin.flow.component.orderedlayout.FlexComponent
-import com.vaadin.flow.component.orderedlayout.HorizontalLayout
+import com.vaadin.flow.component.button.Button
+import com.vaadin.flow.component.icon.VaadinIcon
 
 /**
  * Warning type notification component.
+ *
+ * @param title the warning notification title.
+ * @param message the warning notification message.
  */
-open class VWarningNotification(title: String?, message: String) : VAbstractNotification() {
+open class VWarningNotification(title: String?,
+                                message: String,
+                                locale: String)
+  : VAbstractNotification(title, message, locale) {
 
   //-------------------------------------------------
   // IMPLEMENTATION
   //-------------------------------------------------
 
-  /**
-   * Creates the warning notification footer.
-   */
-  override fun createFooter() {
-    val footer = HorizontalLayout()
-    footer.add(close)
-    footer.isSpacing = true
-    footer.justifyContentMode = FlexComponent.JustifyContentMode.CENTER
-    super.setFooter(footer)
-  }
-
-  override fun setButtons(locale: String) {
-    close = VInputButton(LocalizedProperties.getString(locale, "CLOSE"))
-    close.addClickListener { hide() }
-    close.width = "20%"
-    close.height = "50%"
+  override fun setButtons() {
+    close = Button(LocalizedProperties.getString(locale, "CLOSE"))
+    close.addClickListener { close() }
+    close.isAutofocus = true
+    buttons.add(close)
   }
 
-  override fun onEnterEvent(keyDownEvent: ShortcutEvent?) {
-    TODO("Not yet implemented")
-  }
-
-  override fun onRightEvent(keyDownEvent: ShortcutEvent?) {
-    TODO("Not yet implemented")
-  }
-
-  override fun onLeftEvent(keyDownEvent: ShortcutEvent?) {
-    TODO("Not yet implemented")
-  }
+  override val iconName: VaadinIcon
+    get() = VaadinIcon.EXCLAMATION_CIRCLE
 
   //--------------------------------------------------
   // DATA MEMBERS
   //--------------------------------------------------
-  override val iconName: String?
-    get() = "exclamation-circle"
-
-  private var close = VInputButton()
-
-  //--------------------------------------------------
-  // CONSTRUCTOR
-  //--------------------------------------------------
-  init {
-    super.title = H3(title)
-    super.message = Span(message)
-    super.initialize(title, message, locale)
-  }
+  private lateinit var close: Button
```

Comment:
can we initialize the button here ! = Button(LocalizedProperties.getString(locale, "CLOSE"))